content/en/docs/tasks/manage-gpus/scheduling-gpus.md

```diff
@@ -83,10 +83,52 @@ a different label key if you prefer.
 
 ## Automatic node labelling {#node-labeller}
 
-If you're using AMD GPU devices, you can deploy
-[Node Labeller](https://github.com/RadeonOpenCompute/k8s-device-plugin/tree/master/cmd/k8s-node-labeller).
-Node Labeller is a {{< glossary_tooltip text="controller" term_id="controller" >}} that automatically
-labels your nodes with GPU device properties.
+As an administrator, you can automatically discover and label all your GPU enabled nodes
+by deploying Kubernetes [Node Feature Discovery](https://github.com/kubernetes-sigs/node-feature-discovery) (NFD).
+NFD detects the hardware features that are available on each node in a Kubernetes cluster.
+Typically, NFD is configured to advertise those features as node labels, but NFD can also add extended resources, annotations, and node taints.
+NFD is compatible with all [supported versions](/releases/version-skew-policy/#supported-versions) of Kubernetes.
 
-Similar functionality for NVIDIA is provided by
-[GPU feature discovery](https://github.com/NVIDIA/gpu-feature-discovery/blob/main/README.md).
+Administrators can leverage NFD to also taint nodes with specific features, so that only pods that request those features can be scheduled on those nodes.
+After a cluster is labeled with the GPU feature, you can schedule pods on GPU nodes by adding the following to your pod spec:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: example-vector-add
+spec:
+  # You can then use Kubernetes node affinity to schedule pods on GPU nodes.
+  # https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#node-affinity
+  affinity:
+    nodeAffinity:
+      requiredDuringSchedulingIgnoredDuringExecution:
+        nodeSelectorTerms:
+        - matchExpressions:
+          - key: "gpu.gpu-vendor.example/installed-memory"
+            operator: Gt
+            values: ["40535"]
+          - key: "gpu.gpu-vendor.example/family"
+            operator: In
+            values:  # examples are GCU families not GPU families 😉
+            - arbitrary
+            - armchair-traveller
+            - just-read-the-instructions
+            - steely-glint
+  restartPolicy: Never
+  containers:
+    - name: example-vector-add
+      image: "registry.example/example-vector-add:v42"
+      resources:
+        limits:
+          gpu-vendor.example/example-gpu: 1 # requesting 1 GPU
+      nodeSelector:
+      # Use Kubernetes node selector to schedule pods on GPU nodes.
+      # https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#nodeselector
+        gpu-vendor.example/example-gpu-present: "true" 
+```
+
+#### GPU vendor implementations
+
```

Comment:
Is there an Intel implementation? If so, we should add a link (the entries are sorted alphabetically).

Reply:
K8s-sig/NFD main contributors are INTEL and NVIDIA. For very specific vendored labels NVIDIA uses the gpu-feature-discovery sidecar, which calls to NFD API to generate `nvidia.com/` labels.
For all INTEL features, NFD generates them with `feature.node.kubernetes.io/cpu-cpuid.ADX` as example.

Reply:
is there a GFD version at INTEL @marquiz  ??

Reply:
It's the Intel device plugins for kubernetes (https://github.com/intel/intel-device-plugins-for-kubernetes) that's I think already linked above. Anything to add @eero-t ?

Reply:
Please list it here too, if it works as a plugin to NFD.

Reply:
I'm not sure what the above mentioned "NFD API" or "plugin to NFD" (vendor specific fork of NFD worker?) mean, as NFD is configured through NFD label/etc rules and runtime-provided feature (config) file(s) provided to the per-node NFD workers.

Intel uses following:
* Intel device plugins project has NFD config rules for device plugin DaemonSets: https://github.com/intel/intel-device-plugins-for-kubernetes/tree/main/deployments/nfd/overlays/node-feature-rules/
* Deployed GPU plugin will then provide NFD feature file for additional, more targeted labels [1]: https://github.com/intel/intel-device-plugins-for-kubernetes/blob/main/cmd/gpu_plugin/labels.md

[1] These labels were earlier provided by (deprecated) GPU plugin init container NFD hook binary [2], but hooking support will be removed in next NFD version [3].

[2] https://intel.github.io/intel-device-plugins-for-kubernetes/cmd/gpu_nfdhook/README.html

[3] https://github.com/kubernetes-sigs/node-feature-discovery/issues/856